src/common.c

```diff
@@ -633,7 +846,7 @@ static inline uint32_t read_flash_sr(stlink_t *sl) {
     else if (sl->flash_type == STLINK_FLASH_TYPE_WB)
         sr_reg = STM32WB_FLASH_SR;
     else
-        sr_reg = FLASH_SR;
+        return -1;
```

Comment:
If possible add some error message before exit.

Reply:
I'd like to put some asserts on that kind of unexpected "not implemented" codepaths - ending here with a not supported chip will lead to bad thing and need an a abort. But yes. I'll do that.